### PR TITLE
Fix syslog_syslog_server_info module output, update tests.

### PR DESCRIPTION
**Pull request by anazobec**
Fixed a minor issue pointed out in pull request #132.
> Info module does not convert ``SYSLOG_PROTOCOL_UDP`` to ``udp``, but it should.
> The syslog_server_info module should return values that are same as those accepted by syslog_server module.

### CODE REVIEW

File: plugins/module_utils/syslog_server.py [hunk 1]
```diff
@@ -144,6 +146,9 @@ def get_state(
             )
         ]
 
+        for s in state:
+            s["protocol"] = protocols[s["protocol"]]  # type: ignore
+
```
Comment (justinc1) [inline plugins/module_utils/syslog_server.py:151]: Please move this part of code into `from_hypercore()` method - this is the place to do mapping from hypercore representation into our internal representation.

File: tests/unit/plugins/module_utils/test_syslog_server.py [hunk 2]
```diff
@@ -23,6 +23,9 @@
     reason=f"requires python{MIN_PYTHON_VERSION[0]}.{MIN_PYTHON_VERSION[1]} or higher",
 )
 
+PROTOCOL_TCP = "SYSLOG_PROTOCOL_TCP"
+ANSIBLE_TCP = "tcp"
```
Comment (justinc1) [inline tests/unit/plugins/module_utils/test_syslog_server.py:27]: Variable names like `HYPERCORE_PROTOCOL_TCP` and `ANSIBLE_PROTOCOL_TCP` would be more explicit.

**Review by justinc1 — CHANGES_REQUESTED** [review]
Just small changes. i feel a bit bad for being annoying. Hope is not to bad. :)

File: plugins/module_utils/syslog_server.py (hunk 1 above)
Comment (anazobec) [reply plugins/module_utils/syslog_server.py]: You shouldn't feel bad about it, that part of code was actually not good for my eyes either and knew there should be a better way to go about it, I just couldn't think of it at the moment. :)

**Comment by anazobec** [comment]
Note: I have to fix unit tests. Missed that.

**Review by justinc1 — APPROVED** [review]
Thank you, merging.

**Pull request merged**